src/account.ts

```diff
@@ -159,6 +159,7 @@ export class Account {
             if (error.type === 'TimeoutError') {
                 result = await this.retryTxResult(txHash, this.accountId);
             } else {
+                error.transactionHash = base_encode(txHash);
```

Comment:
Should we move this downstream to provider? Looks like right now we won't catch cases when `this.retryTxResult` throws any other error except `RetriesExceeded`.

Reply:
Or maybe just handle it here, just make sure this situation is covered.

Reply:
Downstream provider doesn't have txnId readily available, so I have a slight preference to keep as and cover the extra cases.